### PR TITLE
Release 1.4.0

### PR DESCRIPTION
**Pull request by fsimonis**
## Step my step guide
* [x] Merge master to develop (No commits after the release on master)
* [x] Do regression tests at develop _list below :arrow_down:_
* [x] Look over CHANGELOG and add things if necessary
* [x] Bump version in CMakeLists.txt, SConstruct, and CHANGELOG.
* [x] Draft message to mailing list @uekerman 
* [x] Update documentation
* [x] Fix potential problems in develop
* [ ] Merge develop to master
* [ ] Tag release (on GitHub) and merge back to develop
* [ ] Send email and do marketing @uekerman 
* [ ] Generate Debian package(s)

## Regression Tests
* [x] SU2/CalculiX
* [x] OpenFOAM
* [x] Alya
* [x] 1D-ElasticTube (C++ and Python)
* [x] SuperMUC
* [x] Fortran Dummy
* [x] C++ Dummy

### CODE REVIEW

**Comment by floli** [comment]
Since I am out of the loop for release discussion, I can't truthfully approve nor reject. Therefore, go for a release whenever you deem it ready!

**Comment by floli** [comment]
Do we create a pre-release, s.t. I can add the AUR PKGBUILD for 1.4 and test it before finalizing?

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,44 @@
...
+    Use `tools/updateSourceFiles.py` from project-root to update all necessary files.
+  - `install` target installs:
+     - the library `PREFIX/lib`.
+     - the binaries `PREFIX/bin` and their manfiles `PREFIX/share/man/man1`.
+     - the cmake configuration files `PREFIX/lib/cmake/precice`.
+     - the necessary files to run testprecice `PREFIX/share/precice`. Use this as `PRECICE_ROOT` on installed system.
+  - CTest definition of tests run in isolated working directories:
+    - `precice.Base` for the base test suite
+    - `precice.MPI2` run on 2 MPI ranks
+    - `precice.MPI4` run on 4 MPI ranks
```
Comment (uekerman) [inline CHANGELOG.md]: @fsimonis This is outdated, right?

Comment (fsimonis) [reply CHANGELOG.md]: Yes the info regarding the tests is outdated

**Comment by fsimonis** [comment]
From my point of view, this is ready for release.

The only problems we still have are CI related:
* the move from SCons to CMake in the system tests has some problems. This is mainly due to the fact that we now run the tests prior to install, which requires mpiexec, which does not like to be run by root.
* the move from SCons to CMake for the general Travis builds in the main repository fails, due to some caching issue with Eigen3.

This can all be done post-release though.

**Comment by BenjaminRueth** [comment]
> the move from SCons to CMake in the system tests has some problems. This is mainly due to the fact that we now run the tests prior to install, which requires mpiexec, which does not like to be run by root.

There is now a user `precice` in all baseimages (https://github.com/precice/systemtests/commit/513ea2258674665d82a24dd4b3dd719e88d112cb). I hope this helps fixing this issue. However, we should also use this user in the actual tests `Test_*`.

**Pull request merged**